### PR TITLE
Clean up README.md

### PR DESCRIPTION
**Pull request by jiachengxu**
**What this PR does / why we need it**:
1. Fix broken links in README.md
2. Add documentation session and point documentation to the documentation website:  https://docs.kubermatic.com/kubecarrier,
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,482 +32,16 @@ KubeCarrier is currently in early development and is not ready for production us
...
-
-Team A is offering the `CouchDB` service from their Kubernetes cluster `eu-west-1` and Team B created an instance of the `CouchDB` service.
-
-Because Team A decided to hide the `.spec.version` property it's absent from the CRD that tenants of Team A have access to. While the internal CRD retains that field, so the provider can use it to orchestrate their workload.
+- [What is KubeCarrier](./docs/_index.md)
+- [Getting Started](./docs/getting_started)
+- [Architecture](./docs/architecture/_index.md)
+- [Repository Organization](./docs/repository_organization/_index.md)
+- [Glossary](./docs/glossary/_index.md)
+- [API Reference](./docs/api_reference/_index.md)
```
Comment (nmiculinic) [inline README.md:44]: I'd delete this, linking to the docs should be cleaner. See for example how the Kubernetes or etcd do it, they just link to the docs and you should go from there. They don't list docs index in the README.md

Comment (jiachengxu) [reply README.md:44]: I list docs index because kubermatic and kubeone also did that, WDYT @thetechnick ?

**Pull request merged**